utils/dbt_format_utils.py

```diff
@@ -0,0 +1,329 @@
+            with open(model_file_path, "r") as model_file:
+                for line in model_file:
+                    # Dont start writing to file if were still in config section
+                    if "-- Final base SQL" in line:
+                        skip_line = False
+                    # For these models, get rid of default where statement
+                    if "where 1 = 1" in line:
+                        skip_line = True
+                    if skip_line:
+                        continue
+                    # Substitute source with 'cta' if regex match
+                    line = re.sub("(?<=source\(').*?(?=', '_airbyte_raw_)", "cta", line)
+                    temp_file.write(line)
+            temp_file.write(partition_replacement_footer)
+            temp_file.seek(0)
+            with open(f"{model_file_path}", "w") as output_file:
+                output_file.write(temp_file.read())
+
+
+# Function to grab the content of a specified cacher sript
+def download_cacher_script(script_guid, api_key, api_token, output_path):
+    # Make API request to get all cacher snippets for a user
+    cacher_url = "https://api.cacher.io/integrations/show_all"
+    headers = {
+        "cache-control": "no-cache",
+        "x-api-key": api_key,
+        "x-api-token": api_token,
+    }
+    resp = requests.get(cacher_url, headers=headers)
+    resp.raise_for_status()
+
+    # Grab all CTA Snippets from user's teams
+    user_teams = resp.json()["teams"]
+    cta_team = next(team for team in user_teams if team.get("name") == "CTA")
+    cta_snippets = cta_team["library"]["snippets"]
+
+    # Find target snippet
+    target_script = next(
+        snippet for snippet in cta_snippets if snippet.get("guid") == script_guid
+    )
+    script_content = target_script.get("files")[0]["content"]
+
+    with open(output_path, "w") as output_file:
+        output_file.write(script_content)
+
+    print(
+        f"Cacher Snippet {target_script.get('title', script_guid)} downloaded to {output_path}"
+    )
+
+
+# Function to append "_base" to all files in a directory
+def add_base_to_filenames(base_tables_path):
+    base_tables = [
+        f
+        for f in os.listdir(base_tables_path)
+        if os.path.isfile(os.path.join(base_tables_path, f))
+    ]
+    base_tables.sort()
+    for table_file in base_tables:
+        table_name = table_file.split(".")[0]
+        # Skip if not sql file
+        if ".sql" not in table_file:
+            print(f"{table_file} not a SQL file. Skipping.")
+            continue
+        # Skip if _base already in filename
+        if "_base.sql" in table_file:
+            print(f"- name: {table_name}")
+            continue
+
+        os.rename(
+            f"{base_tables_path}{table_file}",
+            f"{base_tables_path}{table_name}_base.sql",
+        )
+        print(f"- name: {table_name}_base")
+
+
+def create_matview_dbt_files_from_base(base_tables_path, output_path):
```

Comment:
I can't quickly pinpoint where in the code this should be revised, but I think it will be easier for the matviews to query from {{ ref('TABLE_base') }} instead of {{ source('cta','TABLE_base') }} - the latter strat requires adding the base tables to sources.yml, and we can just skip that step by using ref instead 👀